### ISSUE TITLE
Infinite Buffer Size in /people Topic

### ISSUE THREAD

**Issue by RoboEvangelist** [ipa320/cob_people_perception#52]
Folks,

Check in line 192 of `cob_people_perception/cob_openni2_tracker/ros/src/body_tracker.cpp` where the buffer size is Zero (infinite size). It should be at least 1, as this can cause issue when there is a lot of subscribers for this topic.

**Comment by floweisshardt** [comment]
@ipa-rmb: please check

**Comment by ipa-rmb** [comment]
Thanks for your suggestion. I adapted the code accordingly. Thanks for helping us to improve this software.

Just for curiosity, could you tell us whether you had concrete problems with the infinite queue in the body_tracker.cpp or with some other software? What kind of problems did it cause?

**Comment by RoboEvangelist** [comment]
I just noticed that when I have many nodes subscribing to the /people topic and running for a very long time (72 continues hours tracking a dummy skeleton), then the program would crash eventually. Then I changed the queue size, and the program has not been crashing under the same conditions. Since I have not seen the same error, then I assumed queue size has something to do with it. But I don't have concrete proof, though. What I do know is that if queue size is 0, then memory size can increase indefinitely.
